### PR TITLE
feat: improve api methods of load all

### PR DESCRIPTION
**Pull request by rudigier**

### CODE REVIEW

File: src/services/posts/index.ts [hunk 1]
```diff
@@ -159,86 +161,108 @@ type TGetPostByQuery = {
...
 };
 
-const getPostsByUserId = async ({ id, accessToken }: TGetPostById) => {
+type TGetPostByUserId = {
+	id: string;
+	limit?: number;
+	accessToken: string;
+};
+
+const getPostsOfUser = async ({ id, limit, accessToken }: TGetPostByUserId) => {
```
Comment (flxtagi) [inline src/services/posts/index.ts:184]: suggestion: interface name to `TgetPostOfUser` or rename the method to `getPostsByUserID` . I would prefer the latter. as we have more 'byUser-' methods.

File: src/pages/api/posts/[id]/index.ts [hunk 2]
```diff
@@ -3,8 +3,16 @@ import { NextApiRequest, NextApiResponse } from 'next';
...
+const HTTP_METHODS = {
+	GET: 'GET',
+	POST: 'POST',
+	PUT: 'PUT',
+	DELETE: 'DELETE',
+};
+
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-	const { id } = req.query;
+	const method = req.method;
```
Comment (flxtagi) [inline src/pages/api/posts/[id]/index.ts:14]: I really like this Idea of haveing those HTTP methods dynamicly! 🎉

File: src/services/users/index.ts [hunk 3]
```diff
@@ -35,30 +49,26 @@ const getUsers = async ({ limit, offset = 0, accessToken }: TGetUser): Promise<T
...
 	const users = data.map(transformUser) as TUser[];
 
-	// load more users if limit is not set and the amount of users is less than the total count
-	if (limit === undefined && users.length < count) {
-		const remainingOffset = offset + users.length;
-		const { users: remainingUsers } = await getUsers({
+	// If there are more entries to fetch, make a recursive call
+	if (count > 0 && (!limit || limit > users.length)) {
+		const remainingLimit = limit ? limit - users.length : undefined;
+		const remainingOffset = limit ? offset + limit : offset + users.length;
```
Comment (flxtagi) [inline src/services/users/index.ts:61]: I am not sure if I get that right but this 
 if count = 1 and limit = undefined  => true
 if count = 2 limit = 1 user.length = 0 => true
 if count = 1 limit = 1 user.length = 1  =>false
 
 -> did not test but its brainfu##k - have to test that

**Review by flxtagi — COMMENTED** [review]
Thanks. thats quiete a bit overhaul. Did you adapt all api calls in the pages?

File: src/services/posts/index.ts (hunk 1 above)
Comment (rudigier) [reply src/services/posts/index.ts:184]: then we need to change all the function names: 
	getUser => getUserByUserID
	getPost => getPostByPostID
	deletePost => deletePostByPostID
	getPostsOfUser => getPostsByUserID
	getPostsLikedByUser => getPostsLikedByUserID
	getPostReplies => getPostRepliesByPostID
	
I don't like this long names... do you?

Comment (flxtagi) [reply src/services/posts/index.ts:184]: - true that. and its not really self-explanatory this naming too (getPostsLikedByUserID).. 
I think the most important thing is that we dont mix naming convention TgetPost**Of**User  vs TgetPost**By**User

File: src/services/users/index.ts (hunk 3 above)
Comment (flxtagi) [reply src/services/users/index.ts:61]: ah got that. thanks.

**Review by flxtagi — APPROVED** [review]
nice. let's work with that.

**Pull request merged**